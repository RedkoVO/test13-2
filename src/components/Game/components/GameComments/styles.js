import LikeIcon from '../../../../assets/images/svg/like.svg'
import CommentIcon from '../../../../assets/images/svg/comment.svg'

export default theme => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'flex-start',
    width: '100%',
    paddingLeft: '100px',
    boxSizing: 'border-box',
    marginTop: '50px',
    color: '#fff',

    [theme.breakpoints.down('md')]: {
      paddingLeft: '20px'
    }
  },
  commentItem: {
    position: 'relative',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.52)',
    borderRadius: '25px',
    padding: '20px 30px',
    boxSizing: 'border-box',
    marginBottom: '35px',

    [theme.breakpoints.down('md')]: {
      maxWidth: '740px'
    }
  },
  comment1: {
    width: '700px'
  },
  commentUser: {
    position: 'absolute',
    top: '-15px',
    left: '-7px',
    display: 'block',
    width: '30px',
    height: '30px',
    borderRadius: '50%'
  },
  commentUserPhoto: {
    width: '30px',
    height: '30px'
  },
  commentTitle: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'space-between',
    width: '100%'
  },
  commentTitleText: {
    fontSize: '18px',
    fontWeight: 700
  },
  commentDate: {
    fontSize: '11px',
    fontWeight: 400
  },
  subComment: {
    padding: '7px 10px 7px 35px',
    position: 'absolute',
    borderRadius: '18px',
    bottom: '-15px',
    right: '30px',
    width: '120px',
    boxSizing: 'border-box',
    fontSize: '12px',
    fontWeight: 700,
    color: '#fff',
    textDecoration: 'none',
    backgroundColor: '#000'
  },
  commentLike: {
    padding: '7px 18px 7px 25px',
    position: 'absolute',
    borderRadius: '18px',
    bottom: '-15px',
    right: '165px',
    fontSize: '12px',
    fontWeight: 700,
    color: '#fff',
    textDecoration: 'none',
    backgroundColor: '#000'
  },
  likeValue: {
    position: 'relative',
    color: '#ff4040',

    '&:before': {
      content: `""`,
      position: 'absolute',
      width: '10px',
      height: '10px',
      backgroundImage: `url(${LikeIcon})`,
      backgroundRepeat: 'no-repeat',
      backgroundPosition: '0 0',
      backgroundSize: '10px',
      left: '-13px',
      bottom: '1px'
    }
  },
  commentValue: {
    position: 'relative',

    '&:before': {
      content: `""`,
      position: 'absolute',
      width: '10px',
      height: '10px',
      backgroundImage: `url(${CommentIcon})`,
      backgroundRepeat: 'no-repeat',
      backgroundPosition: '0 0',
      backgroundSize: '10px',
      left: '-15px',
      bottom: '1px'
    }
  },
  comment2: {
    width: '800px'
  },
  comment3: {
    width: '600px'
  }
})
