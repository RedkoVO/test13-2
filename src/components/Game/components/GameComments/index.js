import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import withStyles from '@material-ui/core/styles/withStyles'
import cn from 'classnames'

import styles from './styles'

const GameComments = ({ classes, comments }) => (
  <div className={classes.root}>
    {comments.map(item => (
      <div className={cn(classes.commentItem, classes[`comment${item.id}`])} key={item.id}>
        <Link to="/" className={classes.commentUser}>
          <img src={item.avatar} alt="" className={classes.commentUserPhoto} />
        </Link>
        <div className={classes.commentTitle}>
          <span className={classes.commentTitleText}>{item.title} says:</span>
          <time dateTime="2018-05-22 13:30" className={classes.commentDate}>{item.time}</time>
        </div>
        <p className={classes.commentText}>{item.text}</p>
        <Link to="/" className={classes.commentLike}> <span className={classes.likeValue}>{item.like}</span> Like</Link>
        <Link to="/" className={classes.subComment}><span className={classes.commentValue}>{item.comment}</span> Comment</Link>
      </div>
    ))}
  </div>
)

GameComments.propTypes = {
  classes: PropTypes.object,
  comment: PropTypes.array
}

export default withStyles(styles)(GameComments)