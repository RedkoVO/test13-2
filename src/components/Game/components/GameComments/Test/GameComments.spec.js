import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import GameComments from '../'

const comments = [
  {
    id: 1,
    avatar: 'UserImg',
    title: 'Ben F.',
    time: '22 May 2018 12:03',
    text:
      'Curabitur tincidunt orci quis odio aliquet placerat. Nullam laoreet justo eros, quis semper turpis gravida eget. Ut ullamcorper nibho a massa tincidunt malesuada. In nunc velit basank. s semper turpis gravida eget. Ut ullamcorper nibho a massa tincidunt malesuada. In nunc velit basank.',
    like: 5,
    comment: 7
  },
  {
    id: 2,
    avatar: 'UserImg',
    title: 'Thomas D.',
    time: '22 May 2018 12:03',
    text:
      'Nulla non pellentesque sem, sed aliquet justo. Praesent egestas purus et bibendum tempus.Interdum et malesuada fames ac ante ipsum primis in faucibus.',
    like: 5,
    comment: 7
  }
]

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter keyLength={0}>
        <GameComments comments={comments} />
      </MemoryRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
