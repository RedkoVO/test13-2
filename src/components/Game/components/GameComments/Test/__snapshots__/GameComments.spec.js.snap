// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="GameComments-root-1"
>
  <div
    className="GameComments-commentItem-2 GameComments-comment1-3"
  >
    <a
      className="GameComments-commentUser-4"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="GameComments-commentUserPhoto-5"
        src="UserImg"
      />
    </a>
    <div
      className="GameComments-commentTitle-6"
    >
      <span
        className="GameComments-commentTitleText-7"
      >
        Ben F.
         says:
      </span>
      <time
        className="GameComments-commentDate-8"
        dateTime="2018-05-22 13:30"
      >
        22 May 2018 12:03
      </time>
    </div>
    <p>
      Curabitur tincidunt orci quis odio aliquet placerat. Nullam laoreet justo eros, quis semper turpis gravida eget. Ut ullamcorper nibho a massa tincidunt malesuada. In nunc velit basank. s semper turpis gravida eget. Ut ullamcorper nibho a massa tincidunt malesuada. In nunc velit basank.
    </p>
    <a
      className="GameComments-commentLike-10"
      href="/"
      onClick={[Function]}
    >
       
      <span
        className="GameComments-likeValue-11"
      >
        5
      </span>
       Like
    </a>
    <a
      className="GameComments-subComment-9"
      href="/"
      onClick={[Function]}
    >
      <span
        className="GameComments-commentValue-12"
      >
        7
      </span>
       Comment
    </a>
  </div>
  <div
    className="GameComments-commentItem-2 GameComments-comment2-13"
  >
    <a
      className="GameComments-commentUser-4"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="GameComments-commentUserPhoto-5"
        src="UserImg"
      />
    </a>
    <div
      className="GameComments-commentTitle-6"
    >
      <span
        className="GameComments-commentTitleText-7"
      >
        Thomas D.
         says:
      </span>
      <time
        className="GameComments-commentDate-8"
        dateTime="2018-05-22 13:30"
      >
        22 May 2018 12:03
      </time>
    </div>
    <p>
      Nulla non pellentesque sem, sed aliquet justo. Praesent egestas purus et bibendum tempus.Interdum et malesuada fames ac ante ipsum primis in faucibus.
    </p>
    <a
      className="GameComments-commentLike-10"
      href="/"
      onClick={[Function]}
    >
       
      <span
        className="GameComments-likeValue-11"
      >
        5
      </span>
       Like
    </a>
    <a
      className="GameComments-subComment-9"
      href="/"
      onClick={[Function]}
    >
      <span
        className="GameComments-commentValue-12"
      >
        7
      </span>
       Comment
    </a>
  </div>
</div>
`;
